parl/algorithms/torch/maddpg.py

```diff
@@ -0,0 +1,165 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import parl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from copy import deepcopy
+
+__all__ = ['MADDPG']
+
+from parl.core.torch.policy_distribution import SoftCategoricalDistribution
+from parl.core.torch.policy_distribution import SoftMultiCategoricalDistribution
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Please move this line into the class context.
It may cause incorrect GPU detection: https://github.com/PaddlePaddle/PARL/pull/667#issuecomment-867295465

Reply:
done